backend/plugin/db/redis/sync.go

```diff
@@ -17,38 +18,17 @@ import (
 // SyncInstance syncs the instance metadata.
 func (d *Driver) SyncInstance(ctx context.Context) (*db.InstanceMetadata, error) {
 	var instance db.InstanceMetadata
-	var databaseCount int
-
-	version, err := d.getVersion(ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get server version")
-	}
-	instance.Version = version
-
-	clusterEnabled, err := d.getClusterEnabled(ctx)
+	dbNumber, err := d.getDatabaseNumber(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to check if cluster is enabled")
-	}
-
-	// Redis cluster can only use database zero.
-	if clusterEnabled {
-		databaseCount = 1
-	} else {
-		count, err := d.getDatabaseCount(ctx)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to get databases")
-		}
-		databaseCount = count
+		return nil, errors.Wrap(err, "failed to get database number")
 	}
-
 	var databases []*storepb.DatabaseSchemaMetadata
-	for i := 0; i < databaseCount; i++ {
+	for i := range dbNumber {
```

Comment:
Should be dbNumber[i]?
```suggestion
	for _, n := range dbNumber {
```

Reply:
Could you explain why `i` is used here?

`getDatabaseNumber` returns database numbers, e.g. []int{0, 3, 6}, no?

Reply:
yes, return the db num tha have keys from info command,and display to front.

Reply:
i made a mistake, it should be 
```golang
for _, n := range dbNumber 
```